protected/models/LogicNode.php

```diff
@@ -0,0 +1,17 @@
+<?php
+
+/**
+ * Class LogicNode
+ */
+abstract class LogicNode extends SearchNode
+{
+    /**
+     * @var SearchNode[]
+     */
+    protected $children = array();
+
+    public function addChildren($children)
```

Comment:
This seems to be overriding the children array, not adding to it

Reply:
I haven't quite nailed down how I want the node hierarchy to be constructed. This will likely change as time goes on.

Reply:
Either way this should append the array unless you want to call it setChildren. There should probably also be an addChild function.